packages/core/test/foundry/optimistic-assertor/OptimisticAsserter.Lifecycle.t.sol

```diff
@@ -0,0 +1,37 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.0;
+
+import "../fixtures/optimistic-assertor/OptimisticAssertorFixture.sol";
+import "../fixtures/common/TestAddress.sol";
+
+contract OptimisticAsserterLifecycle is Test {
+    OptimisticAssertor optimisticAssertor;
+    TestnetERC20 defaultCurrency;
+    Timer timer;
+    string claimAssertion = 'q:"The sky is blue"';
+
+    function setUp() public {
+        OptimisticAssertorFixture.OptimisticAsserterContracts memory oaContracts =
+            new OptimisticAssertorFixture().setUp();
+        optimisticAssertor = oaContracts.optimisticAssertor;
+        defaultCurrency = oaContracts.defaultCurrency;
+        timer = oaContracts.timer;
+    }
+
+    function testAssertionWithNoDispute() public {
+        vm.startPrank(TestAddress.account1);
+        defaultCurrency.allocateTo(TestAddress.account1, optimisticAssertor.defaultBond());
+        assert(defaultCurrency.balanceOf(TestAddress.account1) >= optimisticAssertor.defaultBond());
+        defaultCurrency.approve(address(optimisticAssertor), optimisticAssertor.defaultBond());
+
+        bytes32 assertionId = optimisticAssertor.assertTruth(bytes(claimAssertion));
+
+        // Move time forward to the end of the liveness period.
```

Comment:
before time travel we should also check that `settleAndGetAssertion` would revert

Reply:
also catching the reason "Assertion not expired"

Reply:
Good points, added in latest commit